api/appStore/deployment/AppStoreDeploymentNewRestHandler.go

```diff
@@ -0,0 +1,263 @@
+	validator *validator.Validate,
+) *AppStoreDeploymentRestHandlerImpl {
+	return &AppStoreDeploymentRestHandlerImpl{
+		Logger:                    Logger,
+		userAuthService:           userAuthService,
+		enforcer:                  enforcer,
+		enforcerUtil:              enforcerUtil,
+		enforcerUtilHelm:          enforcerUtilHelm,
+		appStoreDeploymentService: appStoreDeploymentService,
+		validator:                 validator,
+	}
+}
+
+func (handler AppStoreDeploymentRestHandlerImpl) CreateInstalledApp(w http.ResponseWriter, r *http.Request) {
+	decoder := json.NewDecoder(r.Body)
+	userId, err := handler.userAuthService.GetLoggedInUser(r)
+	if userId == 0 || err != nil {
+		common.WriteJsonResp(w, err, "Unauthorized User", http.StatusUnauthorized)
+		return
+	}
+	var request appStoreBean.InstallAppVersionDTO
+
+	err = decoder.Decode(&request)
+	if err != nil {
+		handler.Logger.Errorw("request err, CreateInstalledApp", "err", err, "payload", request)
+		common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+		return
+	}
+	err = handler.validator.Struct(request)
+	if err != nil {
+		handler.Logger.Errorw("validation err, CreateInstalledApp", "err", err, "payload", request)
+		common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+		return
+	}
+	token := r.Header.Get("token")
+
+	//rbac block starts from here
+	var rbacObject string
+	if util2.GetDevtronVersion().ServerMode == util2.SERVER_MODE_HYPERION {
+		rbacObject = handler.enforcerUtilHelm.GetHelmObjectByClusterId(request.ClusterId, request.Namespace, request.AppName)
+	}else{
+		rbacObject = handler.enforcerUtil.GetHelmObjectByAppNameAndEnvId(request.AppName, request.EnvironmentId)
+	}
+	if ok := handler.enforcer.Enforce(token, casbin.ResourceHelmApp, casbin.ActionCreate, rbacObject); !ok {
+		common.WriteJsonResp(w, fmt.Errorf("unauthorized user"), nil, http.StatusForbidden)
+		return
+	}
+	//rbac block ends here
+
+	isChartRepoActive, err := handler.appStoreDeploymentService.IsChartRepoActive(request.AppStoreVersion)
+	if err != nil {
+		handler.Logger.Errorw("service err, CreateInstalledApp", "err", err, "payload", request)
+		common.WriteJsonResp(w, err, nil, http.StatusInternalServerError)
+		return
+	}
+	if !isChartRepoActive {
+		common.WriteJsonResp(w, fmt.Errorf("chart repo is disabled"), nil, http.StatusNotAcceptable)
+		return
+	}
+
+	request.UserId = userId
+	handler.Logger.Infow("request payload, CreateInstalledApp", "payload", request)
+	ctx, cancel := context.WithCancel(r.Context())
+	if cn, ok := w.(http.CloseNotifier); ok {
+		go func(done <-chan struct{}, closed <-chan bool) {
+			select {
+			case <-done:
+			case <-closed:
+				cancel()
+			}
+		}(ctx.Done(), cn.CloseNotify())
+	}
+	ctx = context.WithValue(r.Context(), "token", token)
+	defer cancel()
+	res, err := handler.appStoreDeploymentService.InstallApp(&request, ctx)
+	if err != nil {
+		if strings.Contains(err.Error(), "application spec is invalid") {
```

Comment:
better error handling is needed

Reply:
Yes but this logic/code is legacy for argocd. so avoid doing as for now